components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/util/Utils.java

```diff
@@ -451,7 +452,8 @@ public static String getCallbackURL(org.wso2.carbon.identity.recovery.model.Prop
         }
 
         if (StringUtils.isNotBlank(callbackURL)) {
-            URI uri = new URI(callbackURL);
+            String encodeURL = URLEncoder.encode(callbackURL, IdentityRecoveryConstants.UTF_8);
```

Comment:
Can you explain the requirement to encode the value back?

When the `callbackURL` value is assigned, it is decoded. If the value is meant to be encoded, we can remove the decoding part from there.

Reply:
URI constructor does not accept the strings with spaces. There are some scenarios that `callbackURL`  may contains spaces. In order to handle that we can encoded the url here.

Reply:
My question is in line 449 when the variable is assigned with the value, the property value is decoded. So instead of encoding it again, we can remove the decoding part. Is there any reason for not to do so?

Reply:
Addressed